src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpInterop.cs

```diff
@@ -28,16 +28,16 @@ internal static partial class BrowserHttpInterop
         public static partial void AbortResponse(
             JSObject fetchResponse);
 
-        [JSImport("INTERNAL.http_wasm_readable_stream_controller_enqueue")]
-        public static partial void ReadableStreamControllerEnqueue(
-            [JSMarshalAs<JSType.Any>] object pullState,
+        [JSImport("INTERNAL.http_wasm_request_stream_write")]
+        public static partial Task RequestStreamWrite(
+            JSObject requestStream,
             IntPtr bufferPtr,
             int bufferLength);
 
-        [JSImport("INTERNAL.http_wasm_readable_stream_controller_error")]
-        public static partial void ReadableStreamControllerError(
-            [JSMarshalAs<JSType.Any>] object pullState,
-            Exception error);
+        [JSImport("INTERNAL.http_wasm_request_stream_close")]
+        public static partial Task<JSObject> RequestStreamClose(
```

Comment:
This confuses me, could you please explain why we are returning promise here ?

Reply:
This returns the actual `fetchResponse` promise.

So it works like this:

1. The `FetchStream` method returns the `TransformStream`.
2. `CopyToAsync` to the `TransformStream`.
3. Close the `TransformStream` and return the actual `fetchResponse` to read the response from.